### PR TITLE
Ins 105 make wallet domain proxy and factory

### PR DESCRIPTION
**Pull request by ivanshibitov**
**- What I did**
Add Base code for WalletDomain Proxy & Factory

### CODE REVIEW

File: genesis/example/wallet.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	record, err := wd.ChildStorage.Set(wallet)
+	if err != nil {
+		return "", err
+	}
+
+	return record, nil
+}
+
+func (wd *walletDomain) GetWallet(record string) (resolver.Proxy, error) {
+	wallet, err := wd.ChildStorage.Get(record)
```
Comment (megge-dream) [inline genesis/example/wallet.go]: Its better to use `wd.GetChild(record)`

File: genesis/example/wallet_test.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+	_, err := newWalletDomain(nil)
+	assert.EqualError(t, err, "parent must not be nil")
+}
+
+func TestNewWalletDomain(t *testing.T) {
+	parent := &mockParent{}
+	proxy, err := newWalletDomain(parent)
+
+	assert.NoError(t, err)
+	assert.Equal(t, proxy,
```
Comment (megge-dream) [inline genesis/example/wallet_test.go]: In `assert.Equal` second param is expected value, and third - actual value. And fix indents (last should be at next line).

File: genesis/example/wallet_test.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+func TestWalletDomain_GetClassID(t *testing.T) {
+	parent := &mockParent{}
+	wdomain, err := newWalletDomain(parent)
+
+	assert.NoError(t, err)
+	assert.Equal(t, class.WalletDomainID, wdomain.GetClassID())
+}
+
+func TestWalletDomain_GetWallet_NoShuchRecord(t *testing.T) {
```
Comment (megge-dream) [inline genesis/example/wallet_test.go]: May be _No**Such**Record_?

File: genesis/example/wallet.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	return &walletDomain{
+		BaseDomain: *domain.NewBaseDomain(parent, WalletDomainName),
+	}, nil
+}
+
+func (wd *walletDomain) GetClassID() string {
+	return class.WalletDomainID
+}
+
+func (wd *walletDomain) CreateWallet(fc factory.Factory) (string, error) {
```
Comment (megge-dream) [inline genesis/example/wallet.go]: Add comment to this method

File: genesis/example/wallet.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+	record, err := wd.ChildStorage.Set(wallet)
+	if err != nil {
+		return "", err
+	}
+
+	return record, nil
+}
+
+func (wd *walletDomain) GetWallet(record string) (resolver.Proxy, error) {
```
Comment (megge-dream) [inline genesis/example/wallet.go]: Add comment to this method

File: genesis/example/wallet.go (hunk 4 above)
Comment (LinnikD) [inline genesis/example/wallet.go]: In futher implementation you should pass member to this method

File: genesis/example/wallet.go (hunk 5 above)
Comment (LinnikD) [inline genesis/example/wallet.go]: Do we really need GetWallet method? Can i get another person wallet and act with it?

Comment (ivanshibitov) [reply genesis/example/wallet.go]: I add comment to method in interface

File: genesis/example/wallet.go (hunk 4 above)
Comment (ivanshibitov) [reply genesis/example/wallet.go]: I've already done it for interface method

**Pull request merged**